core/story/story_service.go

```diff
@@ -63,6 +63,19 @@ func (s *StoryService) getStoryAndEnrichTitle(commitMessages []string, rawTitle
 		return story, strings.Join([]string{s.formatStoryInPRTitle(story), rawTitle}, " ")
 	}
 
+	if branchTag != "" {
```

Comment:
The method `findStory` is dedicated to finding the Story (other than having it in the `rawTitle`). Today we have `extractCommitFromMessages`, and we are currently implementing `fetchStory`. Could we add there "finding the Story from the branch tag"? (I think before `fetchStory`?)